driverapi/driverapi.go

```diff
@@ -42,6 +42,14 @@ type Driver interface {
 	// Leave method is invoked when a Sandbox detaches from an endpoint.
 	Leave(nid, eid string) error
 
+	// ProgramExternalConnectivity invokes the driver method which does the necessary
+	// programming to allow the external connectivity dictated by the passed options
+	ProgramExternalConnectivity(nid, eid string, options map[string]interface{}) error
```

Comment:
> Libnetwork will still send the exposed port and mapping information to the network driver as it does today. We decided to go toward a deprecation path for this.

At least that was the plan. If current PR is not in line with that, I will take care of it when addressing your other comment(s). Thanks.